test/test_temporary_storage_layout.cu

```diff
@@ -0,0 +1,177 @@
+#include "cub/util_device.cuh"
+#include "test_util.h"
+
+#include <memory>
+
+template <int Items>
+std::size_t GetTemporaryStorageSize(std::size_t (&sizes)[Items])
+{
+  void *pointers[Items]{};
+  std::size_t temp_storage_bytes{};
+  cub::AliasTemporaries(nullptr, temp_storage_bytes, pointers, sizes);
```

Comment:
Check that this returns `cudaSuccess` just to be safe. (no such code path exists when passed `nullptr`, but I'm pedantic :D)